File: cs471/HW/hw1/gcd_full.rb
def gcdI(i, j)
	while i != j
		if i > j
			i = i -j
	  	else
	    	j = j - i
	  	end
	end
	i
end

def gcdF(i, j)
	if j == 0
		i
	else
		gcdF(j, i % j)
	end
end

if ARGV.length != 2 
  puts "gcd_full.rb usage: [m] [n]" 
  exit
end

puts "gcdI #{gcdI(ARGV[0].to_i, ARGV[1].to_i)}"
puts "gcdF #{gcdF(ARGV[0].to_i, ARGV[1].to_i)}"

File: cs471/HW/hw1/gcd_full.py
#! /usr/bin/env python
import sys

def gcdI(i, j):
  while i != j:
  	if (i > j):
  		i = i - j
  	else:
  		j = j - i
  return i

def gcdF(i, j):
	if (j == 0):
		return i
	else:
		return gcdF(j, i % j)

if len(sys.argv) < 3:
  print("%s usage: [I] [J]" % sys.argv[0])
  exit()

print(gcdI(int(sys.argv[1]), int(sys.argv[2])))
print(gcdF(int(sys.argv[1]), int(sys.argv[2])))